src/services/browser/BrowserSession.ts

```diff
@@ -1,36 +1,95 @@
 import * as vscode from "vscode"
 import * as fs from "fs/promises"
 import * as path from "path"
-import { Browser, Page, ScreenshotOptions, TimeoutError, launch } from "puppeteer-core"
+import { exec, spawn } from "child_process"
+import { Browser, Page, ScreenshotOptions, TimeoutError, launch, connect } from "puppeteer-core"
 // @ts-ignore
 import PCR from "puppeteer-chromium-resolver"
 import pWaitFor from "p-wait-for"
 import { setTimeout as setTimeoutPromise } from "node:timers/promises"
+import axios from "axios"
 import { fileExistsAtPath } from "../../utils/fs"
 import { BrowserActionResult } from "../../shared/ExtensionMessage"
 import { BrowserSettings } from "../../shared/BrowserSettings"
-// import * as chromeLauncher from "chrome-launcher"
+import { discoverChromeInstances, testBrowserConnection, isPortOpen } from "./BrowserDiscovery"
+import * as chromeLauncher from "chrome-launcher"
+import { Controller } from "../../core/controller"
+import { telemetryService } from "../../services/telemetry/TelemetryService"
 
 interface PCRStats {
 	puppeteer: { launch: typeof launch }
 	executablePath: string
 }
 
-// const DEBUG_PORT = 9222 // Chrome's default debugging port
+// Define browser connection info interface
+export interface BrowserConnectionInfo {
+	isConnected: boolean
+	isRemote: boolean
+	host?: string
+}
+
+const DEBUG_PORT = 9222 // Chrome's default debugging port
```

Comment:
It would be much better to have a single config object for all hard coded values like DEBUG_PORT and various timeouts everywhere. The timeout objects will tell you why something needs to time out.

Reply:
I agree with this.  If not a config object, then this should at least be defined in a constants file.